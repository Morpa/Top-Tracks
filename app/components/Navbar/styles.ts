import styled, { css } from 'styled-components'

export const Wrapper = styled.nav`
  ${({ theme }) => css`
    padding: ${theme.spacings.xxsmall};
  `}
`

export const Link = styled.a`
  text-decoration: none;
  cursor: pointer;
`
