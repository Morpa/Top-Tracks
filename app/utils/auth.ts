import dotenv from 'dotenv'

dotenv.config()

const state = {
  token: null,
  expires: 0
}

export const getToken = async () => {
  const now = Math.round(new Date().getTime() / 1000)
  const remaining = state.expires - now

  if (remaining <= 60) {
    state.token = null
    state.expires = 0
  }

  if (state.expires === 0) {
    const { access_token, expires_in } = await fetch(
      'https://accounts.spotify.com/api/token',
      {
        method: 'POST',
        headers: {
          Authorization: `Basic ${Buffer.from(
            process.env.SPOTIFY_CLIENT_ID +
              ':' +
              process.env.SPOTIFY_CLIENT_SECRET
          ).toString('base64')}`,
          'Content-Type': 'application/x-www-form-urlencoded'
        },
        body: 'grant_type=client_credentials'
      }
    ).then((response) => response.json())

    state.token = access_token
    state.expires = now + expires_in
  }

  return state.token
}
