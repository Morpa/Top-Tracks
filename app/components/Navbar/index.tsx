import { Heading } from '../Heading'
import * as S from './styles'

export const Navbar = () => (
  <S.Wrapper>
    <Heading size="huge" lineBottom color="white" lineColor="secondary">
      TOP TRACKS
    </Heading>
  </S.Wrapper>
)
