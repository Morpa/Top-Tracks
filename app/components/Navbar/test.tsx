import { MemoryRouter } from 'react-router-dom'
import { render, screen } from '~/utils/test-utils'
import Navbar from '.'

describe('<Navbar />', () => {
  it('should render the heading', () => {
    const { container } = render(<Navbar />, { wrapper: MemoryRouter })

    expect(
      screen.getByRole('img', { name: /frontend-br/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
