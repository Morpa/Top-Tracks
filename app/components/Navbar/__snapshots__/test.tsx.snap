// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render the heading 1`] = `
.c1 {
  color: #FAFAFA;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 5.2rem;
}

.c1::after {
  position: absolute;
  left: 0;
  bottom: -1rem;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #3CD3C1;
}

.c0 {
  padding: 0.8rem;
  cursor: pointer;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<nav
  class="c0"
>
  <h2
    class="c1"
    color="white"
  >
    TOP TRACKS
  </h2>
</nav>
`;
